File: five-little-monkeys.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	numbers := make(map[int]string)
	numbers[1] = "One"
	numbers[2] = "Two"
	numbers[3] = "Three"
	numbers[4] = "Four"
	numbers[5] = "Five"

	var keys []int
	for k := range numbers {
		keys = append(keys, k)
	}
	sort.Sort(sort.Reverse(sort.IntSlice(keys)))

	for _, k := range keys {
		if k == 1 {
			fmt.Printf("%s little monkey jumping on the bed\n", numbers[k])
		} else {
			fmt.Printf("%s little monkeys jumping on the bed\n", numbers[k])
		}

		if k == 1 {
			fmt.Println("They fell off and bumped their head")
		} else {
			fmt.Println("One fell off and bumped their head")
		}

		fmt.Println("Mama called the doctor,")
		fmt.Println("And the doctor said")

		if k == 1 {
			fmt.Println("Put those monkeys right to bed\n")
		} else {
			fmt.Println("No more monkeys jumping on the bed\n")
		}
	}
}

File: five-little-monkeys.pl
use strict;
use warnings;
use 5.010;

my %numbers = (
    1 => 'One',
    2 => 'Two',
    3 => 'Three',
    4 => 'Four',
    5 => 'Five'
);

for my $key ( sort {$b<=>$a} keys %numbers) {
    if ($key == 1) {
        print "$numbers{$key} little monkey jumping on the bed\n";
        print "They fell off and bumped their head\n";
    } else {
        print "$numbers{$key} little monkeys jumping on the bed\n";
        print "One fell off and bumped their head\n";
    }
    
    print "Mama called the doctor,\n";
    print "And the doctor said\n";
    
    if ($key == 1) {
        print "Put those monkeys right to bed\n\n";
    } else {
        print "No more monkeys jumping on the bed\n\n"
    }
}